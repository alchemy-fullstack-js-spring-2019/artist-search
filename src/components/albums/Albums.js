import React from 'react';
import PropTypes from 'prop-types';
import Album from './Album';
import { Link } from 'react-router-dom';

function Albums({ albumArray,  match }) {
  console.log('match', match);
  const albums = albumArray.map((album, i) => {
    let coverArtUrl;
    const hasCoverArt = album['cover-art-archive'].front;
    if(hasCoverArt) {
      coverArtUrl = `http://coverartarchive.org/release/${album.id}/front`;
    } else {
      coverArtUrl = 'https://fakeimg.pl/640x360';
    }
    const albumId = `/${match.params.artistName}/${match.params.artistId}/${album.id}`;
    return (
      <li key={i}>
        <Link to={albumId}>
          <Album album={album.title} image={coverArtUrl} />
        </Link>
      </li>
    );
  });

  return (
    <ul>
      {albums}
    </ul>
  );
}


Albums.propTypes = {
  albumArray: PropTypes.array.isRequired,
  match: PropTypes.object.isRequired
};

export default Albums;
