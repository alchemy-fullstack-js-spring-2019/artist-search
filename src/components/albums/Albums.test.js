import React from 'react';
import { shallow } from 'enzyme';
import Albums from './Albums';

describe('Album component',  ()=>{
  it('renders stuff', ()=>{
    const wrapper = shallow(<Albums albumArray={[{ 'cover-art-archive':true, title:'test' }]} image='test image' match={{ params:'url' }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
