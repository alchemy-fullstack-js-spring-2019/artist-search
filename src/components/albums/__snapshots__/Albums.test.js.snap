// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album component renders stuff 1`] = `
<ul>
  <li
    key="0"
  >
    <Link
      to="/undefined/undefined/undefined"
    >
      <Album
        album="test"
        image="https://fakeimg.pl/640x360"
      />
    </Link>
  </li>
</ul>
`;
